manual/C/ch_Finance-Quote.xml

```diff
@@ -249,6 +249,201 @@ SYNOPSIS
+      <abstract>
+        <para>This program returns quote data for a source and a list of symbols in a format which is easy
+          to read for humans. It is useful for checking that a given online quote source is online and functional. 
+        </para>
+      </abstract>
+
+      <itemizedlist>
+        <listitem>
+          <para>To get a stock price, run a command from the command line of the form:
+            <cmdsynopsis>
+              <command>gnc-fq-dump</command>
+              <arg choice="opt">-v</arg>
+              <arg choice="plain"><replaceable>Source</replaceable></arg>
+              <arg choice="plain" rep="repeat"><replaceable>Symbol</replaceable></arg>
+            </cmdsynopsis>
+            e.g. <command>gnc-fq-dump -v alphavantage IBM</command>
+          </para>
+        </listitem>
+        
+        <listitem>
+          <para>To retrieve currency exchange rate run the following command:
+            <command>gnc-fq-dump <optional>-v</optional> currency USD EUR</command>
+            <footnote>
+              <para>Since &app-fq; 1.41 the default source for currencies is
+                hardcoded to be <quote>Alpha Vantage</quote>. The old yahoo
+                currencies are still available as <command>gnc-fq-dump yahoo_json USDEUR=X</command>.
+              </para>
+            </footnote>
+          </para>
+        </listitem>
+      </itemizedlist>
+
+      <para>To test that &app-fq; is working for currencies within &app;,
+        <orderedlist numeration="arabic">
+          <listitem>
+            <para>find a two-currency transaction and</para>
+          </listitem>
+          <listitem>
+            <para>right-click on it, then</para>
+          </listitem>
+          <listitem>
+            <para>select &gmi.ac.ed-ex; in the context menu.</para>
+          </listitem>
+        </orderedlist>
+        The <guilabel>Transfer</guilabel> dialog will appear. Click the 
+        <guilabel>Fetch Rate</guilabel> button and if everything is
+        behaving itself the current rate will fill in the exchange rate box.
+      </para>
+    </sect2>
+
+    <sect2 id="gnc-fq-helper">
+      <title>gnc-fq-helper</title>
+
+      <abstract>
+        <para>This program is a filter, which does the same thing as gnc-fq-dump, but in a form which
+          the &app; understands. Input and output are Scheme expressions. This helps developers to decide,
+          if the error is in &app; or &app-fq;.
+        </para>
+      </abstract>
+
+      <para>As an example, in September 2008 an Unilever bond was reported to fail. Here is a sample
+        execution and result of this helper program:
+      </para>
+
+      <formalpara>
+        <title>Sample of a real bug in September 2008</title>
+
+        <para>
+<screen language="sh">
+user@host:~$ echo '(yahoo_json "A0GFY7.SG")' | gnc-fq-helper
+(("A0GFY7.SG" (symbol . "A0GFY7.SG") (gnc:time-no-zone . "2008-09-26 16:58:00")
+(last . 89.50) (currency . "EUR ")))
+</screen>
```

Comment:
A 14-year-old bug taken out of context is not very useful.

Reply:
It was thought as an exercise and as I don't know a spoiler tag in mediawikii, I linked the bug for the solution.